contracts/tokenfactory-issuer/src/helpers.rs

```diff
@@ -87,3 +89,40 @@ pub fn check_is_not_frozen(deps: Deps, denom: &str) -> Result<(), ContractError>
 
     Ok(())
 }
+
+pub fn create_metadata(
+    denom: String,
+    AdditionalMetadata {
+        description,
+        denom_units,
+        display,
+        name,
+        symbol,
+    }: AdditionalMetadata,
+) -> Metadata {
+    let has_base_denom_units = denom_units.iter().any(|unit| unit.denom == denom);
+    Metadata {
+        denom_units: if has_base_denom_units {
+            denom_units
+        } else {
+            prepend_default_base_denom_unit(denom.clone(), denom_units)
+        },
```

Comment:
We should probably just error if the base denom unit is not included correctly, instead of leaving this much flexibility